src/components/Settings/SettingsPlex.tsx

```diff
@@ -52,12 +62,143 @@ interface SyncStatus {
   libraries: Library[];
 }
 
+interface PresetServerDisplay {
+  name: string;
+  ssl: boolean;
+  uri: string;
+  address: string;
+  host: string;
+  port: number;
+  local: boolean;
+}
 interface SettingsPlexProps {
   onComplete?: () => void;
 }
 
+function PlexPresetOption(props: { key: string; server: PresetServerDisplay }) {
+  if (props.server.local) {
+    return (
+      <option key={props.server.uri} value={props.server.uri}>
+        {' '}
+        {props.server.name}
+        {' ('}
+        {props.server.address}
+        {') [local]'}
+      </option>
+    );
+  }
+  return (
+    <option key={props.key} value={props.server.uri}>
+      {' '}
+      {props.server.name}
+      {' ('}
+      {props.server.address}
+      {') [remote]'}
+    </option>
+  );
+}
+
 const SettingsPlex: React.FC<SettingsPlexProps> = ({ onComplete }) => {
   const intl = useIntl();
+  const { addToast, removeToast } = useToasts();
+  const availablePresets = (): Map<string, PresetServerDisplay> => {
+    const map: Map<string, PresetServerDisplay> = new Map<
+      string,
+      PresetServerDisplay
+    >();
+    // turns the URI into a usable hostname / IP
+    const regexp = /(http(s?):\/\/)(.*)(:[0-9]*)/;
+    dataServers?.forEach((dev) => {
+      if (dev.provides.includes('server') && dev.owned == true) {
+        dev.connection.forEach((conn) => {
+          const theHost = conn.uri.replace(regexp, '$3');
+          map.set(conn.uri, {
+            name: dev.name,
+            ssl: conn.protocol == 'https' ? true : false,
+            uri: conn.uri,
+            address: conn.address,
+            port: conn.port,
+            local: conn.local,
+            host: theHost,
+          });
+        });
+      }
+    });
+    return map;
+  };
```

Comment:
You are working with dataServers here before its defined. This only works because we call the method /after/ dataServers is set, but it's very confusing.

You should define this method (and the following ones) under your `useSWR` hook and work on the data directly when the state actually changes. You should also use `useMemo` so we only run through the array again if the data is actually different by setting `dataServers` as a dependency. You should just also return the map directly, instead of a function.